Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -121,57 +129,55 @@ void Digitizer::flush(DigitContainer& digits, o2::dataformats::MCTruthContainer<
 
 SignalContainer Digitizer::addSignalsFromPileup()
 {
-  int count = 0;
   SignalContainer addedSignalsMap;
+  int nSignalsToRemove = 0;
   for (const auto& collection : mPileupSignals) {
+    bool pileupSignalBecomesObsolete = false;
     for (int det = 0; det < MAXCHAMBER; ++det) {
       const auto& signalMap = collection[det]; //--> a map with active pads only for this chamber
       for (const auto& signal : signalMap) {   // loop over active pads only, if there is any
         const int& key = signal.first;
         const SignalArray& signalArray = signal.second;
         // check if the signal is from a previous event
         if (signalArray.firstTBtime < mCurrentTriggerTime) {
-          if ((mCurrentTriggerTime - signalArray.firstTBtime) < BUSY_TIME) {
+          pileupSignalBecomesObsolete = true;
+          if ((mCurrentTriggerTime - signalArray.firstTBtime) > READOUT_TIME) { // OS: READOUT_TIME should actually be drift time (we want to ignore signals which don't contribute signal anymore at mCurrentTriggerTime)
             continue; // ignore the signal if it  is too old.
           }
           // add only what's leftover from this signal
           // 0.01 = samplingRate/1000, 1/1000 to go from ns to micro-s, the sampling rate is in 1/micro-s
           int idx = (int)((mCurrentTriggerTime - signalArray.firstTBtime) * 0.01); // number of bins to add, from the tail
-          auto it0 = signalArray.signals.begin() + idx;
+          auto it0 = signalArray.signals.begin() + (constants::TIMEBINS - idx);
```

Comment:
I think this should be:
```
auto it0 = signalArray.signals.begin() + idx;
```
Otherwise, the above should be defined:
```
int idx = (int)((mCurrentTriggerTime - signalArray.firstTBtime) * 0.01 - constants::TIMEBINS); // number of bins to add, from the tail
```